experimental/generation/generator/packages/library/templates/generator.lg

```diff
@@ -74,275 +74,252 @@
+
+# actionSetOrPushOperation(value)
+> Set or push value based on singelton or array
+- IF: ${isArrayProperty(property)}
+    - ```
+    {
+        "$kind": "Microsoft.EditArray",
+        "changeType": "push",
+        "itemsProperty": "$${property}",
+        "value": "=${value}"
+    }```
+- ELSE:
+    -```
+    {
+        "$kind": "Microsoft.SetProperty",
+        "property": "$${property}",
+        "value": "=${value}"          
+    }```
+
+# actionDeletePropertyToChange()
+> Delete propertyToChange
 - ```
 {
-    "$schema": "${appSchema}",
-    "$kind": "Microsoft.OnAssignEntity",
-    "operation": "Add()",
-    "property": "${property}",
-    "entity": "${entity}",
+   "$kind": "Microsoft.IfCondition",
+    "condition": "$PropertyToChange == '${property}'",
     "actions": [
+        {
+            "$kind":"Microsoft.DeleteProperty",
+            "property": "$PropertyToChange"
+        }
+    ]
+}```
+
+# verifyPattern(entity, else)
+> Verify entity matches pattern
+- IF: ${propertySchema.pattern}
+    - ```
+    {
+        "$kind": "Microsoft.IfCondition",
+        "condition":"not(isMatch( @${entity},  dialogClass.schema.properties['${property}'].pattern))",
+        "actions":[
+            {
+                "$kind": "Microsoft.SendActivity",
+                "activity": "\${${templateName(property)}_NoPatternMatch(@${entity})}"
+            }                    
+        ],
+        "elseActions":[${else}]
+    }```
+- ELSE:
+    - ${else}
+
+# verifyMin(entity, else)
+> Verify entity minimum value
+- IF: ${propertySchema.minimum}
+    -```
         {
             "$kind": "Microsoft.IfCondition",
             "condition":"@${entity} < dialogClass.schema.properties['${property}'].minimum",
             "actions":[
                 {
                     "$kind": "Microsoft.SendActivity",
-                    "activity": "\${${property}_IsTooSmall(@${entity})}"
+                    "activity": "\${${templateName(property)}_IsTooSmall(@${entity})}"
                 }                    
             ],
-            "elseActions":[
-                {
-                    "$kind": "Microsoft.IfCondition",
-                    "condition":"@${entity} > dialogClass.schema.properties['${property}'].maximum",
-                    "actions":[
-                        {
-                            "$kind": "Microsoft.SendActivity",
-                            "activity": "\${${property}_IsTooBig(@${entity})}"
-                        }                    
-                    ],
-                    "elseActions": [
-                        {
-                            "$kind": "Microsoft.SendActivity",
-                            "activity": "\${confirmationText('${property}',  @${entity})}"
-                        },
-                        ${setOrPush()}
-                    ]
-                }
-            ]
-        }
-    ]               
-    ${designerOpEntity('Add', entity)}     
-}
-```
+            "elseActions":[${else}]
```

Comment:
I am not sure if drafting the dialog in this way, could minimum and maximum appear in the same dialog ?